packages/utils/telemetry-utils/src/config.ts

```diff
@@ -0,0 +1,335 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ITelemetryBaseLogger, ITelemetryLogger } from "@fluidframework/common-definitions";
+import { Lazy } from "@fluidframework/common-utils";
+import { DebugLogger } from "./debugLogger";
+import { ChildLogger, ITelemetryLoggerPropertyBags } from "./logger";
+
+export type ConfigTypes = string | number | boolean | number[] | string[] | boolean[] | undefined;
+
+/**
+ * Base interface for providing configurations to enable/disable/control features
+ */
+export interface IConfigProviderBase {
+    getRawConfig(name: string): ConfigTypes;
+}
+
+/**
+ * Explicitly typed interface for reading configurations
+ */
+ export interface IConfigProvider extends IConfigProviderBase {
+    getBoolean(name: string, defaultValue?: boolean): boolean | undefined;
```

Comment:
@andre4i i wonder if it would be better to require the defaultValue, but allow undefined
like `getBoolean(name: string, defaultValue: boolean | undefined): boolean | undefined`

this would encourage developers to specify a default value, as they will need to explicitly specify undefined otherwise, rather than just leaving it off.

its a small change but i think we do want to encourage default values.

Reply:
I agree.